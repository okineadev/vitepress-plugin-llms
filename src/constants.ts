/** Default template for the `llms.txt` file. */
export const defaultLLMsTxtTemplate = `\
# {title}

{description}

{details}

## Table of Contents

{toc}`

/** List of unnecessary files grouped by category. */
export const unnecessaryFilesList = {
	/** `index.md` */
	indexPage: ['index.md'],
	/** Blogs */
	blogs: ['blog/*', 'blog.md'],
	/** Pages about team */
	team: ['team.md'],
} as const satisfies Record<string, readonly string[]>
