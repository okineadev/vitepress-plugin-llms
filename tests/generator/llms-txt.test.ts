import { afterAll, describe, expect, it, mock } from 'bun:test'
import path from 'node:path'

import { defaultLLMsTxtTemplate } from '@/constants'

import mockedFs from '../mocks/fs'

mock.module('node:fs/promises', () => mockedFs)

const { readFile } = mockedFs.default

import {
	generateLLMsTxt,
	// @ts-ignore
} from '@/generator/llms-txt'
import { fakeCustomLlmsTxtTemplate, outDir, preparedFilesSample } from '../resources'
import fakeIndexMd from '../test-assets/index.md'

describe('generateLLMsTxt', () => {
	readFile.mockReturnValue(Promise.resolve(fakeIndexMd))
	afterAll(() => readFile.mockReset())

	it('generates a `llms.txt` file', async () => {
		expect(
			await generateLLMsTxt(preparedFilesSample.slice(1), {
				indexMd: path.join(outDir, 'index.md'),
				outDir: outDir,
				LLMsTxtTemplate: defaultLLMsTxtTemplate,
				templateVariables: {},
				vitepressConfig: {},
			}),
		).toMatchSnapshot()
	})

	it('works correctly with base config of vitepress', async () => {
		expect(
			await generateLLMsTxt(preparedFilesSample.slice(1), {
				indexMd: path.join(outDir, 'index.md'),
				outDir: outDir,
				LLMsTxtTemplate: defaultLLMsTxtTemplate,
				templateVariables: {},
				vitepressConfig: {
					base: '/docs/',
				},
			}),
		).toMatchSnapshot()
	})

	it('works correctly with a custom template', async () => {
		expect(
			await generateLLMsTxt(preparedFilesSample.slice(1), {
				indexMd: path.join(outDir, 'index.md'),
				outDir: path.join(outDir),
				LLMsTxtTemplate: fakeCustomLlmsTxtTemplate,
				templateVariables: {},
				vitepressConfig: {},
			}),
		).toMatchSnapshot()
	})
	it('works correctly with a custom template variables', async () => {
		expect(
			await generateLLMsTxt(preparedFilesSample, {
				indexMd: path.join(outDir, 'index.md'),
				outDir: outDir,
				LLMsTxtTemplate: defaultLLMsTxtTemplate,
				templateVariables: { title: 'foo', description: 'bar', toc: 'zoo' },
				vitepressConfig: {},
			}),
		).toMatchSnapshot()
	})

	it('works correctly with a custom template and variables', async () => {
		expect(
			await generateLLMsTxt(preparedFilesSample, {
				indexMd: path.join(outDir, 'index.md'),
				outDir: outDir,
				LLMsTxtTemplate: '# {foo}\n\n**{bar}**\n\n{zoo}',
				templateVariables: { title: 'foo', description: 'bar', toc: 'zoo' },
				vitepressConfig: {},
			}),
		).toMatchSnapshot()
	})
})
