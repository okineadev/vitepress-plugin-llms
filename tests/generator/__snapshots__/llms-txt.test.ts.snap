// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`generateLLMsTxt generates a \`llms.txt\` file 1`] = `
"# Some cool tool

> Blazing fast frontend tool

## Table of Contents

- [Getting started](/test/getting-started.md): Instructions on how to get started with the tool
- [Quickstart](/test/quickstart.md): Instructions for quick project initialization
- [Some other section](/test/other.md)
"
`;

exports[`generateLLMsTxt works correctly with base config of vitepress 1`] = `
"# Some cool tool

> Blazing fast frontend tool

## Table of Contents

- [Getting started](/docs/test/getting-started.md): Instructions on how to get started with the tool
- [Quickstart](/docs/test/quickstart.md): Instructions for quick project initialization
- [Some other section](/docs/test/other.md)
"
`;

exports[`generateLLMsTxt works correctly with a custom template 1`] = `
"# Custom title

> Custom description

## TOC

- [Getting started](/test/getting-started.md): Instructions on how to get started with the tool
- [Quickstart](/test/quickstart.md): Instructions for quick project initialization
- [Some other section](/test/other.md)
"
`;

exports[`generateLLMsTxt works correctly with a custom template variables 1`] = `
"# foo

> bar

## Table of Contents

zoo"
`;

exports[`generateLLMsTxt works correctly with a custom template and variables 1`] = `
"# {foo}

**{bar}**

{zoo}"
`;
